### ISSUE TITLE
Cluster-admin on management cluster for customers

### ISSUE THREAD

**Issue by teemow** [giantswarm/roadmap#1711]
We approaching a point where it makes sense to think about how we can enable cluster-admin for customers on the management cluster. Obviously this should be considered as an option for an advanced platform team. And we need to think about rail guards and responsibilities. But this would allow our customers to build their own platform solutions without handcuffs as the management cluster is a central place for the platform teams to offer services to their development teams.

Pre-requirement: We need to sort out the scope of our secrets on the management:
 * https://github.com/giantswarm/giantswarm/issues/22385
 * https://github.com/giantswarm/giantswarm/issues/20106

Example: https://github.com/giantswarm/roadmap/issues/948#issuecomment-1285222448

**Comment by kuosandys** [comment]
@weatherhog would like to take care of following up

**Comment by teemow** [comment]
@weatherhog we talked about this story in the honey badger planning session and @gianfranco-l will hand this over to you.

**Comment by gianfranco-l** [comment]
the plan is for honeybadger to write down some process docs first and then hold a [knowledge sharing session](https://github.com/giantswarm/roadmap/issues/1725) with Rainbow (specifically about the knowledge sharing, there was an informal chat between the two teams TFs already)

**Comment by teemow** [comment]
For me this is less a technical question but a mindset question. Customers have full access to workload clusters too. Should they use this? No. Can they use this. Yes. Workload clusters are for their development teams. 

The management cluster does "the platform management" and the platform team of the customer is working with the management cluster. But they are often limited and have to find workarounds. 

In general we've always seen us as enablers and not as a managed service. We did'nt want to stand in the way of our customers. This is why we've given full access to the workload clusters although customers can break them. The questions @mproffitt wrote down apply to workload clusters too. Still we gave them admin access.

This is why I'd like us to review the shared responsibility on management clusters. Why are we so afraid of platform teams breaking the management cluster? Why aren't we afraid that customers break workload clusters. And how often will they do this? Imo the freedom for the customer (gs not standing in the way) and us learning about the customers use cases (to adapt our own offering) outweights the risks.

The proposed gitops solution is imo falling too short. Customers need access to debug their services. They need to be able to restart their services. They should learn and see how our services are running etc.
